robotools/evotools/commands.py

```diff
@@ -139,10 +141,12 @@ def prepare_evo_aspirate_dispense_parameters(
         raise ValueError(f"Invalid wells: wells and tips need to have the same length.")
     if labware_position is None:
         raise ValueError("Missing required parameter: position")
-    if not all(isinstance(position, int) for position in labware_position) or any(
-        position < 0 for position in labware_position
-    ):
-        raise ValueError(f"Invalid position: {labware_position}")
+    grid, site = labware_position
+    if not isinstance(grid, int) or not 1 <= grid <= 67:
+        raise ValueError("Grid (first number in labware_position tuple) has to be an int from 1 - 67.")
+    if not isinstance(site, int) or not 1 <= site <= 128:
+        raise ValueError("Site (second number in labware_position tuple) has to be an int from 1 - 128.")
+    labware_position = (grid, site - 1)
```

Comment:
Why this? Because you didn't want to change the rest of the API, where you slice with the sites starting at 0?

Reply:
Internally (`dibecs_contrib`) we use 1-based site numbers and the change here is to align the APIs to the version that users are most familiar with: Tecan sites numbering 1, 2, ...

I decided to make the change in the `prepare_*_parameters` functions that also do the range checks

Reply:
I meant specifically the part of site -1 instead of using the new 1-based site number. This is to be able to slice with it in other use cases?

Reply:
Within commands which are produced by e.g. `evo_aspirate()`, the location of a labware still has to be sepcified with the 0-based system, the 1-based system is just the outward facing one.

Reply:
Yep, then I got it and approve the changes :)

Reply:
Ah no, this is because the `f`-string for the EVOware command still needs the 0-based version.

And the return values of this preparation-function directly go into that `f`-string

Reply:
That's what I meant